services/streammanager/kafka/client/producer.go

```diff
@@ -117,5 +131,20 @@ func (p *Producer) Publish(ctx context.Context, msgs ...Message) error {
 			Headers: headers,
 		}
 	}
-	return p.writer.WriteMessages(ctx, messages...)
+
+	var err error
+	for i := 0; i < p.config.MaxRetries; i++ {
```

Comment:
why aren't we using the `Writer.MaxAttempts` which uses [backoff](https://github.com/segmentio/kafka-go/blob/main/writer.go#L1089)? The default `Writer.MaxAttempts` is 10, so we are actually using it, so why do we need another retry mechanism?

Reply:
Done :+1: